mathics/builtin/box/layout.py

```diff
@@ -305,6 +305,13 @@ def apply_list(self, boxes, evaluation):
         result = RowBox(*items)
         return result
 
+    @property
+    def elements(self):
```

Comment:
It feels like something in the code or the API is wrong here. 

I get nervous about "properties" doing possibly unbounded calculations such as `self.to_expressions()`.  

If I understand correctly, the caller somehow knows this is a compound Expression/Element and assumes that therefore it has an `.element` attribute which is of type `Expression`? 

It feels like what should be happening that the _caller_ should be asking instead for  a _conversion_, that is, should be calling `to_expression().elements` 

I need to think about this more.

Reply:
> It feels like something in the code or the API is wrong here.
> 

Probably it is, but I tried to build over the existing API. At least, this is "localized" in this class.

> I get nervous about "properties" doing possibly unbounded calculations such as `self.to_expressions()`.
> 
> If I understand correctly, the caller somehow knows this is a compound Expression/Element and assumes that therefore it has an `.element` attribute which is of type `Expression`?
> 

The problem is that  `Part` assumes that an Expression has a property `elements`. This just provides the property by creating an Expression. A different possibility would be to implement a different rule for working with RowBoxes. I thought this change was simpler, but maybe there is a better approach.

> It feels like what should be happening that the _caller_ should be asking instead for a _conversion_, that is, should be calling `to_expression().elements`
> 
> I need to think about this more.

Sure.